app/services/catalog/order_state_transition.rb

```diff
@@ -0,0 +1,34 @@
+module Catalog
+  class OrderStateTransition
+    attr_reader :state
+
+    def initialize(order_id)
+      @order = Order.find(order_id)
+    end
+
+    def process
+      @state = determine_order_state
+      @order.update!(:state => @state)
+
+      self
+    end
+
+    private
+
+    def determine_order_state
+      item_states = @order.order_items.collect(&:state)
+
+      if item_states.include?('Failed')
+        'Failed'
+      elsif item_states.include?('Denied')
```

Comment:
@lindgrenj6 I am on the fence if the order should have a state called denied. I understand it being in the Order Item.
If there are multiple order items 1 gets approved and the 2nd one gets denied. We would mark the order as Denied as opposed to "Failed" They could drill down the order items and why it failed.

Reply:
Updated to mark as failed instead of Denied.